### PR TITLE
TiDBMonitor add component label

### LINKED ISSUE #3607: TidbMonitor lost labels
## Bug Report

**What did you do?**
TidbMonitor lost labels includes namespaceLabel ,componentLabel etc...

**What did you expect to see?**
Metric add namespaceLabel ,componentLabel .
**What did you see instead?**
Lost Labels.

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
 Fix  monitor data lost component label.
 Closes #3607 

### What is changed and how does it work?
Optimize prometheus configuration.
### Code changes

- [x] Has Go code change

### Tests

- [x] Unit test 
- [x] E2E test 
- [x] Manual test 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
`component` label is added to the scrape jobs of Prometheus in TiDBMonitor and tidbmonitor does not support multi-cluster monitoring
```

### CODE REVIEW

**Comment by mikechengwei** [comment]
image

**Comment by mikechengwei** [comment]
image
image

File: pkg/monitor/monitor/monitor_manager.go [hunk 1]
```diff
@@ -65,9 +65,9 @@ func (m *MonitorManager) SyncMonitor(monitor *v1alpha1.TidbMonitor) error {
 	if monitor.DeletionTimestamp != nil {
 		return nil
 	}
-	if monitor.Spec.Clusters == nil || len(monitor.Spec.Clusters) < 1 {
-		err := fmt.Errorf("tm[%s/%s] does not configure the target tidbcluster", monitor.Namespace, monitor.Name)
-		return err
+	if monitor.Spec.Clusters == nil || len(monitor.Spec.Clusters) != 1 {
+		klog.Errorf("tm[%s/%s] does not configure the target tidbcluster or contain multiple target tidbclusters ", monitor.Namespace, monitor.Name)
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/monitor_manager.go:69]:
```suggestion
		klog.Errorf("tm[%s/%s] does not configure the target tidbcluster or contain multiple target tidbclusters, ingore sync", monitor.Namespace, monitor.Name)
```

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by mikechengwei** [comment]
Target job  screenapt:
image

**Pull request merged**